claims.go

```diff
@@ -400,34 +320,47 @@ func (p *Claims) validateHwVersion() error {
 	return nil
 }
 
-func (p *Claims) validateNonce() error {
-	if p.Nonce == nil {
-		return fmt.Errorf("missing mandatory nonce")
+func (c *Claims) validateNonce(expectedProfile string) error {
+	var nonce []byte
+	switch expectedProfile {
+	case PSA_PROFILE_1:
+		if c.LegacyNonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		nonce = *c.LegacyNonce
+	case PSA_PROFILE_2:
+		if c.Nonce == nil {
+			return fmt.Errorf("missing mandatory nonce")
+		}
+		if c.Nonce.Len() != 1 {
+			return errors.New("there must be exactly one nonce")
```

Comment:
Any specific reason why Line 328 & 333 are fmt.Errof(), while line 336 is errors.New()

I am just checking the consistency aspect?

Reply:
well spotted, thanks!  Fixed in 721d8b9